go/vt/vtgate/semantics/analyzer_test.go

```diff
@@ -1628,39 +1628,106 @@ func fakeSchemaInfo() *FakeSI {
 	return si
 }
 
+var parentTbl = &vindexes.Table{
```

Comment:
nitpick: maybe time to break out the FK tests into their own testfile?

Reply:
Done